files/image_config/logrotate/templates/logrotate-syslog.j2

```diff
@@ -0,0 +1,29 @@
+{% from "logrotate-common.j2" import postrotate with context %}
+{% from "logrotate-common.j2"
+    import disk_percentage_treshold_rotation with context -%}
+{% from "defaults.j2" import def_logging -%}
+
+{% set syslog = (LOGGING | d({})).get('syslog', {}) -%}
+{% set frequency = syslog.frequency | d(def_logging.common.frequency) -%}
+{% set max_number = syslog.max_number | d(def_logging.syslog.max_number) -%}
+
+{% if syslog.disk_percentage and not syslog.size -%}
+    {% set size_mb = (max_logs_size_kb | int) * 
+                     (syslog.disk_percentage | float) / 100 / 1024 -%}
+{% else -%}
+    {% set size_mb = (syslog.size | d(common_file_size_mb)) | float -%}
+{%- endif %}
+{% set size = (size_mb * 1024 * 1024) | int -%}
+/var/log/syslog
+{
+    {{ frequency }}
```

Comment:
Can you clarify what does this frequency control? The frequency of logrotate itself is being controlled by timer. Why is frequency needed here?

Reply:
this is weekly, daily, monthly etc. https://linux.die.net/man/8/logrotate#:~:text=Each%20log%20file%20may%20be%20handled%20daily%2C%20weekly%2C%20monthly%2C%20or%20when%20it%20grows%20too%20large.